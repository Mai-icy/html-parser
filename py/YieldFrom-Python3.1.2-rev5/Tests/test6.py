#
#   Test delegation of send()
#

def g1():
	print("Starting g1")
	x = yield "g1 ham"
	print("g1 received", x)
	yield from g2()
	x = yield "g1 eggs"
	print("g1 received", x)
	print("Finishing g1")

def g2():
	print("Starting g2")
	x = yield "g2 spam"
	print("g2 received", x)
	x = yield "g2 more spam"
	print("g2 received", x)
	print("Finishing g2")

g = g1()
y = next(g)
x = 1
try:
	while 1:
		y = g.send(x)
		print("Yielded", y)
		x += 1
except StopIteration:
	pass
