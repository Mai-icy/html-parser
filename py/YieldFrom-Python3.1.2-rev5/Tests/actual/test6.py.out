Starting g1
g1 received 1
Starting g2
Yielded g2 spam
g2 received 2
Yielded g2 more spam
g2 received 3
Finishing g2
Yielded g1 eggs
g1 received 4
Finishing g1
